### ISSUE TITLE
PermissionUtils: import java.nio.file unknown to Android Java

### ISSUE THREAD

**Issue by syslogic** [kamranzafar/jtar#21]
Can only tell that this is the reason why it can't work; for reference: http://developer.android.com/reference/java/nio/package-summary.html
On Android 6.0 the permissions are meanwhile obtained per-application, while the effective permissions depend on the mode in which the output stream was opened (that /storage/emulated/0/ is an abstraction layer): http://developer.android.com/guide/topics/data/data-storage.html#filesExternal

**Comment by rodrigoGA** [comment]
+1

**Comment by acuna-personal** [comment]
Have this proble too. Author is inactive since May 2017. Seems I should fork it and add this libraries...

**Comment by syslogic** [comment]
https://developer.android.com/reference/java/nio/file/Files.html#createTempDirectory(java.lang.String, java.nio.file.attribute.FileAttribute<?>...) says: "added in API level 26". Before that, there only is https://developer.android.com/reference/java/io/File.html#createTempFile(java.lang.String, java.lang.String, java.io.File)

**Comment by syslogic** [comment]
for example:

    @Before
    public void setup() throws IOException {
        dir = Files.createTempDirectory("tartest").toFile();
        dir.mkdirs();
    }

should be something alike:

	@Before
	public void setup() throws IOException {
		
		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
			dir = java.nio.file.Files.createTempDirectory(this.prefix).toFile();
		} else {
			/* BUT java.io.File only has a method createTempFile() */
			dir = java.io.File.createTempDirectory(this.prefix).toFile();
		}
		assertTrue(dir.mkdirs());
	}

**Comment by acuna-personal** [comment]
No-no-no, that's wrong. PermissionUtils.java file regulates a POSIX permissions, not Android read/write permissions, because this jTar library was written for `native` Java, so there's no ANY Android compatible code, please don't confuse it. Android supported POSIX **since** API 26 (**not before**), but not fully, there's such a stubs in many its methods, so there's no need to narrow supported Android versions for API 26 and higher, so I've cut the POSIX support from the code, now it's working like a sharm, but I won't make the pull request in this project, because now it's separated project specially for Android. Have you interesting in it?

**Comment by syslogic** [comment]
well, I can write to `/data/user/0/org.kamranzafar.jtar.test/cache` ... by `Context.getCacheDir()`.

**Comment by acuna-personal** [comment]
It's because you've using Android API, not because jTar supports Android. It's simple native Java code which is using some native Java wrappers, not Android, you can simply use it with your Android code, and as it's a native Java code, you can work with it via Android in your project, but can't work with Android directly from its code. But there's no need for you because you have an Android project where you can simply include this library and use it in your project without touching its code.
